### PR TITLE
[Internal] Fixes: `System.Net.Http` by bumping to pinned `4.3.4` reference

### PR DESCRIPTION
**Pull request by seesharprun**
I'm building the `Microsoft.Azure.Cosmos` library with .NET 9 which now shows warnings that your library has the following vulnerabilities in some projects (not all).

- `Newtonsoft.Json` versions **< 13.0.1** (https://github.com/advisories/GHSA-5crp-9r3c-p9vr)
  - This is already handled in #4839 
- `System.Net.Http` versions **< 4.3.4** (https://github.com/advisories/GHSA-7jgj-8wvc-jh57)

I'm not sure if this is an allowed update.

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/Microsoft.Azure.Cosmos.Encryption.Custom.csproj [hunk 1]
```diff
@@ -55,6 +55,10 @@
     <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" PrivateAssets="All" />
     <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.0.102" PrivateAssets="All" />
   </ItemGroup>
+	
+  <ItemGroup>
```
Comment (Pilchie) [inline Microsoft.Azure.Cosmos.Encryption.Custom/src/Microsoft.Azure.Cosmos.Encryption.Custom.csproj:59]: Why a separate ItemGroup?

**Review by Pilchie — COMMENTED** [review]
@kirankumarkolli - can you take a look at this?

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/Microsoft.Azure.Cosmos.Encryption.Custom.csproj (hunk 1 above)
Comment (seesharprun) [reply Microsoft.Azure.Cosmos.Encryption.Custom/src/Microsoft.Azure.Cosmos.Encryption.Custom.csproj:59]: I saw this pattern in a few other .csproj. It's not intentional and I can add it to the previous `ItemGroup` If you want.

Comment (seesharprun) [reply Microsoft.Azure.Cosmos.Encryption.Custom/src/Microsoft.Azure.Cosmos.Encryption.Custom.csproj]: They have been combined in latest.

**Comment by kirankumarkolli** [comment]
On lates master branch its building with just 'Newtonsoft.Json' warning (which we are currently working on fixing it).
@seesharprun are you still seeing issues on latest master?

```
e:\src\v31\Microsoft.Azure.Cosmos.Encryption\src>dotnet build
Restore succeeded with 1 warning(s) in 1.3s
    e:\src\v31\Microsoft.Azure.Cosmos.Encryption\src\Microsoft.Azure.Cosmos.Encryption.csproj : warning NU1903: Package 'Newtonsoft.Json' 10.0.2 has a known high severity vulnerability, https://github.com/advisories/GHSA-5crp-9r3c-p9vr
You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  Microsoft.Azure.Cosmos.Encryption succeeded with 2 warning(s) (4.0s) → bin\Debug\netstandard2.0\Microsoft.Azure.Cosmos.Encryption.dll
    e:\src\v31\Microsoft.Azure.Cosmos.Encryption\src\Microsoft.Azure.Cosmos.Encryption.csproj : warning NU1903: Package 'Newtonsoft.Json' 10.0.2 has a known high severity vulnerability, https://github.com/advisories/GHSA-5crp-9r3c-p9vr
    CSC : warning SA0001: XML comment analysis is disabled due to project configuration (https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA0001.md)

Build succeeded with 3 warning(s) in 7.2s

e:\src\v31\Microsoft.Azure.Cosmos.Encryption\src>dotnet --list-sdks
3.1.426 [C:\Program Files\dotnet\sdk]
6.0.321 [C:\Program Files\dotnet\sdk]
6.0.427 [C:\Program Files\dotnet\sdk]
7.0.120 [C:\Program Files\dotnet\sdk]
8.0.403 [C:\Program Files\dotnet\sdk]
9.0.100-rc.1.24452.12 [C:\Program Files\dotnet\sdk]

e:\src\v31\Microsoft.Azure.Cosmos.Encryption\src>
```

**Comment by kirankumarkolli** [comment]
`System.Net.Http` is a transitive dependency through Azure.Core. 
Earlier we had older Azure.Core version as dependency resulting in above warning, now we upgraded to latest version and that fixed the issue.

**Comment by seesharprun** [comment]
My issue is now resolved. I'll close this request

**Comment by kirankumarkolli** [comment]
Thank you @seesharprun.

**Pull request closed without merging**